pkgs/build-support/build-fhs-userenv/chrootenv.c

```diff
@@ -20,56 +21,112 @@
 #include <sys/stat.h>
 #include <sys/wait.h>
 
-char *env_whitelist[] = {"TERM",
-                         "DISPLAY",
-                         "XAUTHORITY",
-                         "HOME",
-                         "XDG_RUNTIME_DIR",
-                         "LANG",
-                         "SSL_CERT_FILE",
-                         "DBUS_SESSION_BUS_ADDRESS"};
-
-char **env_build(char *names[], size_t len) {
-  char *env, **ret = malloc((len + 1) * sizeof(char *)), **ptr = ret;
-
-  for (size_t i = 0; i < len; i++) {
-    if ((env = getenv(names[i]))) {
-      if (asprintf(ptr++, "%s=%s", names[i], env) < 0)
-        errorf(EX_OSERR, "asprintf");
+#define LEN(x) sizeof(x) / sizeof(*x)
+
+char *env_blacklist[] = {};
+
+char **env_filter(char *envp[]) {
+  char **filtered_envp = malloc(sizeof(*envp));
+  size_t n = 0;
+
+  while (*envp != NULL) {
+    bool blacklisted = false;
+
+    for (size_t i = 0; i < LEN(env_blacklist); i++) {
+      if (!strncmp(*envp, env_blacklist[i], strlen(env_blacklist[i]))) {
```

Comment:
Because I only need to partially match the string. Example of valid entry in `env_blacklist` would be `PWD=`, example of `*envp` that would be blacklisted is `PWD=/home/user`.

Reply:
The second parameter should be the index of the first '=' in the string indeed.

Reply:
@dezgeg I don't think I understand. Could you elaborate?

Reply:
Brain fart of me, sorry.

Reply:
Yeah, the point was that `env_blacklist` elements would always end with an equal sign. But `unsetenv` approach is much better. I've discovered `environ` only today.